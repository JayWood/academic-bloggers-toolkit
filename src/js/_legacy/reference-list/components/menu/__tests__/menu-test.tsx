import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import UIStore from '_legacy/stores/ui/reference-list';
import Menu from '..';

const setup = () => {
    const citationStyle: ABT.CitationStyle = top.ABT.styles.styles[0];
    const data = { citationStyle };
    const store = new UIStore(false);
    const onSubmit = jest.fn();
    const component = shallow(
        <Menu data={data} ui={store} onSubmit={onSubmit} />,
    );
    return {
        component,
        onSubmit,
    };
};

describe('<Menu />', () => {
    it('should match snapshots', () => {
        const { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should handle clicks', () => {
        const { component, onSubmit } = setup();
        const button = component.find('Button').first();
        const submitData = {
            kind: 'OPEN_IMPORT_DIALOG',
        };
        button.simulate('click', {
            currentTarget: { id: 'OPEN_IMPORT_DIALOG' },
        });
        expect(onSubmit).toHaveBeenCalledWith(submitData);
    });
    it('should handle style change', () => {
        const { component, onSubmit } = setup();
        const styleInput = component.find('StyleInput');
        const suggestion: ABT.CitationStyle = {
            kind: 'predefined',
            label: 'American Medical Association',
            value: 'american-medical-association',
        };
        const data = { suggestion };
        const submitData = { kind: 'CHANGE_STYLE', data: data.suggestion };
        styleInput.simulate('selected', {}, data);
        expect(onSubmit).toHaveBeenCalledWith(submitData);
    });
});
