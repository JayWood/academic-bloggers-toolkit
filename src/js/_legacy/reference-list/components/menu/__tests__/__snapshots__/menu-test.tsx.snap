// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshots 1`] = `
<div
  className="menuContainer"
>
  <div
    className="subpanel"
  >
    <Button
      flat={true}
      icon="media-code"
      id="OPEN_IMPORT_DIALOG"
      label="Import references"
      onClick={[Function]}
      tooltip={
        Object {
          "position": "bottom",
          "text": "Import references",
        }
      }
      type="button"
    />
    <Button
      flat={true}
      icon="image-rotate"
      id="REFRESH_PROCESSOR"
      label="Refresh reference list"
      onClick={[Function]}
      tooltip={
        Object {
          "position": "bottom",
          "text": "Refresh reference list",
        }
      }
      type="button"
    />
    <Button
      flat={true}
      icon="trash"
      id="DESTROY_PROCESSOR"
      label="Delete all references"
      onClick={[Function]}
      tooltip={
        Object {
          "position": "bottom",
          "text": "Delete all references",
        }
      }
      type="button"
    />
    <Button
      disabled={false}
      flat={true}
      icon="list-view"
      id="INSERT_STATIC_BIBLIOGRAPHY"
      label="Insert static publication list"
      onClick={[Function]}
      tooltip={
        Object {
          "position": "bottom",
          "text": "Insert static publication list",
        }
      }
      type="button"
    />
    <Button
      flat={true}
      href="https://github.com/dsifford/academic-bloggers-toolkit/wiki"
      icon="editor-help"
      label="Usage instructions"
      role="link"
      tooltip={
        Object {
          "position": "bottom",
          "text": "Usage instructions",
        }
      }
      type="button"
    />
  </div>
  <StyleInput
    currentStyle={
      Object {
        "kind": "predefined",
        "label": "American Medical Association",
        "value": "american-medical-association",
      }
    }
    onSelected={[Function]}
    styles={
      Array [
        Object {
          "kind": "predefined",
          "label": "American Medical Association",
          "value": "american-medical-association",
        },
        Object {
          "kind": "custom",
          "label": "Test Custom Style",
          "value": "test-custom-style",
        },
      ]
    }
  />
</div>
`;
